promql/engine.go

```diff
@@ -727,6 +727,9 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 				s.Point.T = ts
 				mat[i] = Series{Metric: s.Metric, Points: []Point{s.Point}}
 			}
+			if mat.ContainsSameLabelset() {
```

Comment:
You should check result, rather than doing this in two places.

Reply:
The tests are failing if I am just checking `result`.
Errors are being caught when the `combined result` is checked.